apps/couchbase.go

```diff
@@ -319,11 +320,32 @@ func (lp LoggingProcessorCouchbaseHTTPAccess) Components(tag string) []fluentbit
 			},
 		}.Components(tag, lp.Type())...,
 	)
-	c = append(c, confgenerator.LoggingProcessorModifyFields{
+	mf := confgenerator.LoggingProcessorModifyFields{
 		Fields: map[string]*confgenerator.ModifyField{
 			InstrumentationSourceLabel: instrumentationSourceValue(lp.Type()),
 		},
-	}.Components(tag, lp.Type())...)
+	}
+	// Generate the httpRequest structure.
+	for _, field := range []string{
+		"remoteIp",
+		"requestMethod",
+		"requestUrl",
+		// "protocol",
```

Comment:
I was noticing both protocol and referer were missing from the regex. 

I came up with this regex and from my testing I think it captures these fields that are commented out

```regex
^(?<http_request_remoteIp>[^ ]*) (?<host>[^ ]*) (?<user>[^ ]*) \[(?<timestamp>[^\]]*)\] "(?<http_request_requestMethod>\S+) (?<http_request_requestUrl>\S+) (?<http_request_protocol>\S+)" (?<http_request_status>[^ ]*) (?<http_request_responseSize>[^ ]*\S+) (?<http_request_referer>[^ ]*) "(?<http_request_userAgent>[^ ].*)" (?<message>.*)$

```

I think this one should work more universally 😄 
https://regex101.com/r/e0Il6t/1


<details>
<summary>Log Lines:</summary>

```
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-2RMBTYUU.js HTTP/1.1" 200 755016 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 3
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-HSSTCYXX.js HTTP/1.1" 200 8233 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 0
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-G2DN52YO.js HTTP/1.1" 200 78880 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-3I2V4QFF.js HTTP/1.1" 200 147644 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-DIBLBDP7.js HTTP/1.1" 200 7391 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-2FMELT24.js HTTP/1.1" 200 15754 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 0
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-YF7HOUTV.js HTTP/1.1" 200 7692 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-EEVACM5E.js HTTP/1.1" 200 105841 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 2
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-4WPXME7X.js HTTP/1.1" 200 170384 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-ZC47CG5T.js HTTP/1.1" 200 353974 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 2
69.176.133.154 - - [05/Aug/2022:22:16:56 +0000] "GET /jsout/chunk-OWAHUOU4.js HTTP/1.1" 200 1181 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /_p/ui/query/angular-directives/qw.directives.css HTTP/1.1" 200 6562 "http://34.136.63.96:8091/ui/index.html" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /_p/ui/query/ui-current/query.css HTTP/1.1" 200 7029 "http://34.136.63.96:8091/ui/index.html" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /_p/ui/query/angular-directives/qw.json.chart.css HTTP/1.1" 200 269 "http://34.136.63.96:8091/ui/index.html" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /apple-touch-icon.png HTTP/1.1" 200 1136 "http://34.136.63.96:8091/ui/index.html" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /favicon-16x16.png HTTP/1.1" 200 389 "http://34.136.63.96:8091/ui/index.html" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /jsout/mn_overview_controller-G7CMIA4Z.js HTTP/1.1" 200 66560 "http://34.136.63.96:8091/jsout/main.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /jsout/chunk-GB3KXULB.js HTTP/1.1" 200 11552 "http://34.136.63.96:8091/jsout/mn_overview_controller-G7CMIA4Z.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /jsout/chunk-OO6YLJR3.js HTTP/1.1" 200 3133 "http://34.136.63.96:8091/jsout/mn_overview_controller-G7CMIA4Z.js" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
69.176.133.154 - - [05/Aug/2022:22:16:57 +0000] "GET /pools HTTP/1.1" 200 504 "http://34.136.63.96:8091/ui/index.html" "Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0" 1
```
</details>

Let me know if you concur!

Reply:
We should also look to see if maybe we can potentially harden the regex in `genericAccessLogParser` found in [common logging processors](https://github.com/GoogleCloudPlatform/ops-agent/blob/1c13c82135fda19e8343afbeea2eca5a219238fe/apps/common_logging_processors.go#L34) in the future as that one has all the fields already, but I think making an isolated change to couchbase would probably be safer at the moment!